### PR TITLE
Update the Regex section of strings.md.

### PR DESCRIPTION
**Pull request by lewisl**
This adds a brief description and example of using the Regex constructor to create regex strings programmatically.  I could not find this in documentation, but found Regex in source code.

### CODE REVIEW

File: doc/src/manual/strings.md [hunk 1]
```diff
@@ -936,6 +936,33 @@ ERROR: syntax: invalid escape sequence
 Triple-quoted regex strings, of the form `r"""..."""`, are also supported (and may be convenient
 for regular expressions containing quotation marks or newlines).
 
+The Regex() constructor may be used to create a valid regex string programmatically.  This permits using the contents of string variables and other string operations when constructing the regex string. Any of the regex codes above can be used within the single string argument to Regex(). Here are some examples:
```
Comment (KristofferC) [inline doc/src/manual/strings.md]:
```suggestion
The `Regex()` constructor may be used to create a valid regex string programmatically.  This permits using the contents of string variables and other string operations when constructing the regex string. Any of the regex codes above can be used within the single string argument to `Regex()`. Here are some examples:
```

File: doc/src/manual/strings.md [hunk 2]
```diff
@@ -936,6 +936,33 @@ ERROR: syntax: invalid escape sequence
...
+
+```julia-repl
```
Comment (KristofferC) [inline doc/src/manual/strings.md]:
```suggestion
```jldoctest
```

File: doc/src/manual/strings.md [hunk 3]
```diff
@@ -936,6 +936,33 @@ ERROR: syntax: invalid escape sequence
...
+julia> using Dates
+
+julia> d = Date(1962,7,10)
+1962-07-10
+
+julia> regex_d = Regex("Day " * string(day(d)))
+r"Day 10"
+
+julia> match(regex_d,"It happened on Day 10")
```
Comment (KristofferC) [inline doc/src/manual/strings.md]:
```suggestion
julia> match(regex_d, "It happened on Day 10")
```

**Comment by StefanKarpinski** [comment]
@lewisl, note that you should be able to just accept @KristofferC's suggestions directly via GitHub UI.

**Comment by lewisl** [comment]
Hope I did that right:  committed the 3 changes.

**Pull request merged**